src/main/java/de/tum/in/www1/artemis/web/rest/metis/conversation/OneToOneChatResource.java

```diff
@@ -61,6 +63,9 @@ public OneToOneChatResource(OneToOneChatAuthorizationService oneToOneChatAuthori
     public ResponseEntity<OneToOneChatDTO> startOneToOneChat(@PathVariable Long courseId, @RequestBody List<String> otherChatParticipantLogins) throws URISyntaxException {
         var requestingUser = userRepository.getUserWithGroupsAndAuthorities();
         log.debug("REST request to create one to one chat in course {} between : {} and : {}", courseId, requestingUser.getLogin(), otherChatParticipantLogins);
+        if (!(courseRepository.isMessagingEnabled(courseId))) {
+            throw new ResponseStatusException(HttpStatus.FORBIDDEN, "Messaging is not enabled for this course");
+        }
```

Comment:
Out of scope for this PR: Since this check is repeated in several classes, I think it would make sense to refactor it in the future together with some of the other repeated checks into some separate `ConversionAccessChecker` component (I’m not familiar with this part of the Artemis code, maybe something similar exists already).

Reply:
Good idea!